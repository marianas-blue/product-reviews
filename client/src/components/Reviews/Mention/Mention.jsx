/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/no-multi-comp */
/* eslint-disable no-unused-vars */
/* eslint-disable react/button-has-type */
/* eslint-disable no-useless-constructor */
/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import { Title, Container } from './Mention.style';

class Buttons extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      color: true,
      fontColor: true,
    };
    this.changeColor = this.changeColor.bind(this);
  }

  changeColor() {
    this.setState({
      color: !this.state.color,
      fontColor: !this.state.fontColor,
    });
  }

  render() {
    const bgColor = this.state.color ? '#d6e8ea' : '#00464f';
    const txtColor = this.state.fontColor ? 'black' : 'white';
    const style = {
      backgroundColor: bgColor,
      color: txtColor,
      margin: '10px 10px 10px 0px',
      lineHeight: '29px',
      padding: '0px 14px',
      fontSize: '13px',
      outline: 'none',
      boxShadow: 'none',
    };
    const randomArr = ['sound quality', 'battery life', 'charging case',
      'wireless earbuds', 'noise cancelling', 'highly recommend',
      'noise cancellation', 'rose gold', 'listening to music', 'definitely recommend',
      'left earbud', 'left and right'];
    return (
      <div>
        <button
          style={style}
          onClick={this.changeColor}
        >
          {randomArr[Math.floor(Math.random() * 11) + 1]}
        </button>
      </div>
    );
  }
}

class Mention extends React.Component {
  render() {
    return (
      <div>
        <Title>Read reviews that mention</Title>
        <Container>
          <Buttons />
          <Buttons />
          <Buttons />
          <Buttons />
        </Container>
        <Container>
          <Buttons />
          <Buttons />
          <Buttons />
          <Buttons />
        </Container>
        <Container>
          <Buttons />
          <Buttons />
          <Buttons />
          <Buttons />
        </Container>
      </div>
    );
  }
}

export default Mention;
