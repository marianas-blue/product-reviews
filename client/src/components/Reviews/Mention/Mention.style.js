import styled from 'styled-components';

export const Title = styled.button`
font-weight: 700;
font-size: 17px;
line-height: 1.255;
border: none; 
padding: 0;
background-color: transparent;
outline: none;
text-align: left;
`;

export const Container = styled.div`
display: flex;
align-items: row;
align-content: space-around;
`;
