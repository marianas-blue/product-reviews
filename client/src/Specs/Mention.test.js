/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';

import Mention from '../components/Reviews/Mention/Mention';

describe('Mention', () => {
  it('should render correctly', () => {
    const component = shallow(<Mention />);

    expect(component).toMatchSnapshot();
  });
});
