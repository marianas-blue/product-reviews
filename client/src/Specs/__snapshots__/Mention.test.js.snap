// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mention should render correctly 1`] = `
<div>
  <styled.button>
    Read reviews that mention
  </styled.button>
  <styled.div>
    <Buttons />
    <Buttons />
    <Buttons />
    <Buttons />
  </styled.div>
  <styled.div>
    <Buttons />
    <Buttons />
    <Buttons />
    <Buttons />
  </styled.div>
  <styled.div>
    <Buttons />
    <Buttons />
    <Buttons />
    <Buttons />
  </styled.div>
</div>
`;
